### PR TITLE
New stripe gateway implementation

### LINKED ISSUE #3875: Successful transaction in stripe does not populate cc payment fields
### What I'm trying to achieve
Get card details (ex. last 4 digits) from Payment object when using stripe gateway.

### Steps to reproduce the problem
1. Use stripe gateway
2. Pay with card
3. `cc_last_digits`, `cc_brand`, `cc_exp_month`, `cc_exp_month` are empty in Payment object.

### What I expected to happen
Those data should be taken from `gateway_response`.

### LINKED ISSUE #3943: Rewrite stripe integration to use Payment Intent
From Stripe docs:

> Starting in September 2019, a new regulation called Strong Customer Authentication (SCA) will require businesses in Europe to request additional authentication for online payments. We recommend that all European businesses start building their Stripe integrations with either our Payment Intents API or our new Checkout to be ready for these rule changes.

Our current stripe gateway works on old api, which means we need to update it ASAP. 

Docs: https://stripe.com/docs/payments/payment-intents

### PR DESCRIPTION
**Pull request by salwator**
Closes #3943
Fixes #3875

This PR reimplements Stripe gateway integration so it works on new [Payment Intent API](https://stripe.com/docs/api/payment_intents). The main reason for that is [SCA](https://stripe.com/guides/strong-customer-authentication) regulation, which will result in the current gateway stop working soon. 
Additionally, the new implementation introduces better testing using VCR recorded real-life Stripe communication (in test mode) and the ability to store and reuse customers card.

**Current legacy checkout in old saleor storefront will stop working** with stripe due to legacy checkout used, which is not compatible with new Stripe API. Since it will stop working either way due to SCA regulation, old forms are dropped in this PR. 

- [x] Achieve old implementation API feature parity
- [x] Implement 3D secure flow
- [x] Add ability to store customers card info
- [x] Use stored customers card for checkout
- [x] List available stored cards
- [x] Add ability to fetch card details from the gateway and store it in payment object
- [x] Remove old implementation and replace with a new one

---

1. [x] Privileged views and APIs are guarded by proper permission checks.
1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [x] Database migration files are up to date.
1. [x] The changes are tested.
1. [ ] The code is documented (docstrings, project documentation).
1. [x] GraphQL schema and type definitions are up to date.
1. [x] Changes are mentioned in the changelog.

### CODE REVIEW

**Comment by django-queries** [comment]
Here is the report for e58d42c75beda60121d30dc94026ac6f877e8f32 (mirumee/saleor @ new-stripe-gateway)
Base comparison is 3a50f4e5dadeaceb7706a8b01e7f1c754ecdd81c.



<details><summary>No differences were found. (click me)</summary>
<p>

```diff
# api.benchmark checkout
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  add billing address to checkout     	         34	         34	             20
  add shipping to checkout            	          7	          7	              0
  checkout payment charge             	         14	         14	              0
  complete checkout                   	          6	          6	              0
  create checkout                     	         48	         48	             24

# api.benchmark homepage
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  retrieve main menu                  	          5	          5	              0
  retrieve product list               	          4	          4	              0
  retrieve secondary menu             	          5	          5	              0
  retrieve shop                       	          2	          2	              0

# api.benchmark product
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  product details                     	         13	         13	              3

# api.benchmark variant
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  retrieve variant list               	          9	          9	              2
```

</p>
</details>

**Comment by NyanKiyoshi** [comment]
@maarcingebala don't forget to squash merge 😉

**Comment by NyanKiyoshi** [comment]
Will need rebasing once https://github.com/mirumee/saleor/pull/4660 is merged.

File: tests/gateway/stripe/test_stripe.py [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+@pytest.mark.integration
+@pytest.mark.vcr(filter_headers=["authorization"])
+def test_authorize(sandbox_gateway_config, stripe_payment):
+    payment_info = create_payment_information(
+        stripe_payment, PAYMENT_METHOD_CARD_SIMPLE
+    )
+    response = authorize(payment_info, sandbox_gateway_config)
+    assert not response.error
+    assert response.kind == TransactionKind.CAPTURE
+    assert isclose(response.amount, TRANSACTION_AMOUNT)
```
Comment (korycins) [inline tests/gateway/stripe/test_stripe.py]: You know the currency of `TRANSACTION_AMOUNT` so you can quantize a decimal value. 
```
CENTS = Decimal("0.01")
TRANSACTION_AMOUNT.quantize(CENTS)
```

**Comment by salwator** [comment]
> @maarcingebala don't forget to squash merge 😉

I did rebased PR commit history to clean it up and squash related changes.

File: tests/gateway/stripe/test_stripe.py (hunk 1 above)
Comment (salwator) [reply tests/gateway/stripe/test_stripe.py]: Good to know, I've based assertions in new tests on previous test suites. Might be a good idea to refactor this in the near future.

**Pull request merged**